test/int_primary_datasets.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+// this file contains logic for primarydatasets and primarydstypes API
+//
+// primarydatasets:
+// the HTTP request body is defined by dbs.PrimaryDatasetRecord struct defined in dbs/primarydatasets.go
+// the HTTP response body is defined by dbs.PrimaryDataset struct defined in this dbs/primarydatasets.go
+//
+// primarydstypes:
+// the HTTP response body is defined by primaryDSTypesReponse struct defined in this dbs/primarydstypes.go
+//
+// the HTTP handlers and endpoints are defined in the EndpointTestCase struct defined in test/integration_cases.go
+
+import (
+	"net/http"
+	"net/url"
+	"testing"
+
+	"github.com/vkuznet/dbs2go/dbs"
+	"github.com/vkuznet/dbs2go/web"
+)
+
+// expected response from primarydstypes GET API
+type primaryDSTypesResponse struct {
+	DATA_TYPE          string `json:"data_type"`
+	PRIMARY_DS_TYPE_ID int64  `json:"primary_ds_type_id"`
+}
+
+// primarydataset and primarydstype endpoints tests
+func getPrimaryDatasetTestTable(t *testing.T) EndpointTestCase {
+	// create data structs for expected requests and responses
+	primaryDSReq := dbs.PrimaryDatasetRecord{
+		PRIMARY_DS_NAME: TestData.PrimaryDSName,
+		PRIMARY_DS_TYPE: "test",
+		CREATE_BY:       "tester",
+	}
+	primaryDSResp := dbs.PrimaryDataset{
```

Comment:
there is no `dbs.PrimaryDataset` in current codebase, may be you added it in your fork and forget to add to this commit.